batch/build-batch-worker-image-startup-gcp.sh

```diff
@@ -41,12 +41,40 @@ curl -fsSL "https://github.com/GoogleCloudPlatform/docker-credential-gcr/release
   | tar xz --to-stdout ./docker-credential-gcr \
 	> /usr/bin/docker-credential-gcr && chmod +x /usr/bin/docker-credential-gcr
 
+curl -fsSL https://nvidia.github.io/libnvidia-container/gpgkey \
+     | gpg --dearmor -o /usr/share/keyrings/nvidia-container-toolkit-keyring.gpg
+curl -s -L https://nvidia.github.io/libnvidia-container/ubuntu22.04/libnvidia-container.list | \
+    sed 's#deb https://#deb [signed-by=/usr/share/keyrings/nvidia-container-toolkit-keyring.gpg] https://#g' | \
+    tee /etc/apt/sources.list.d/nvidia-container-toolkit.list
+
+# don't use build-essentials, it pulls in gcc-11
+# the latest nvidia driver doesn't build with gcc-11
+#apt-get --yes install gcc-12 g++-12
+apt-get update
```

Comment:
You only need to run update once after installing the Nvidia key (the curl commands).